services/motion/explore/explore.go

```diff
@@ -0,0 +1,626 @@
+// Package explore implements a motion service for exploration.
+package explore
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/edaniels/golog"
+	geo "github.com/kellydunn/golang-geo"
+	"github.com/pkg/errors"
+	servicepb "go.viam.com/api/service/motion/v1"
+	goutils "go.viam.com/utils"
+
+	"go.viam.com/rdk/components/base"
+	"go.viam.com/rdk/components/base/kinematicbase"
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/internal"
+	"go.viam.com/rdk/motionplan"
+	"go.viam.com/rdk/operation"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/robot/framesystem"
+	"go.viam.com/rdk/services/motion"
+	"go.viam.com/rdk/services/vision"
+	"go.viam.com/rdk/spatialmath"
+	"go.viam.com/rdk/utils"
+)
+
+var (
+	model                             = resource.DefaultModelFamily.WithModel("explore")
+	errUnimplemented                  = errors.New("unimplemented")
+	defaultObstaclePollingFrequencyHz = 2.
+	// Places a limit on how far a potential move action can be performed.
+	moveLimit = 10000.
+	// The distance a detected obstacle can be from a base to trigger the Move command to stop.
+	validObstacleDistanceMM = 1000.
+)
+
+func init() {
+	resource.RegisterDefaultService(
+		motion.API, model,
+		resource.Registration[motion.Service, *Config]{
+			Constructor: NewExplore,
+			WeakDependencies: []internal.ResourceMatcher{
+				internal.ComponentDependencyWildcardMatcher,
+			},
+		})
+}
+
+const (
+	exploreOpLabel = "explore-motion-service"
+)
+
+// inputEnabledActuator is an actuator that interacts with the frame system.
+// This allows us to figure out where the actuator currently is and then
+// move it. Input units are always in meters or radians.
+type inputEnabledActuator interface {
+	resource.Actuator
+	referenceframe.InputEnabled
+}
+
+// obstacleDetectorName is used for sending and receiving all obstacle detectors until the motionCfg can be
+// added to the move service.
+type obstacleDetectorName struct {
+	VisionService resource.Name
+	Camera        resource.Name
+}
+
+// obstacleDetectorPair provides a map for matching vision services to any and all cameras to be used with them.
+type obstacleDetectorPair map[vision.Service][]camera.Camera
```

Comment:
[note] After speaking with @nfranczak it seems like we may be moving towards allowing multiple cameras to be associated with a single vision service. I have made this change here. However, the motionCfg still sends over a one-to-one pairing. I have made this code support multiple potential cameras but have not made the change to send over multiple. Let me know if you'd like me to make those changes to navigation builtIn and affected files